src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -364,6 +371,155 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
+                        // all arguments up to and including the last argument needed by this interpolated string conversion into temps, in order
+                        // to ensure we're keeping lexical ordering of side effects.
+                        ensureTempTrackingSetup(ref temps, ref positionsAssignedToTemp);
+                        Debug.Assert(!positionsAssignedToTemp.IsNull);
+
+                        foreach (var placeholder in interpolationData.ArgumentPlaceholders)
+                        {
+                            // Replace each needed placeholder with a sequence of store and evaluate the temp.
+                            var argIndex = placeholder.ArgumentIndex;
+
+                            BoundLocal local;
+                            switch (argIndex)
+                            {
+                                case BoundInterpolatedStringArgumentPlaceholder.InstanceParameter when receiverStored:
+                                    Debug.Assert(visitedReceiver != null && requiresInstanceReceiver);
+                                    local = (BoundLocal)((BoundSequence)visitedReceiver).Value;
+                                    break;
+
+                                case BoundInterpolatedStringArgumentPlaceholder.InstanceParameter:
+                                    Debug.Assert(visitedReceiver != null && requiresInstanceReceiver);
+                                    local = _factory.StoreToTemp(visitedReceiver, out var store, refKind: visitedReceiver.GetRefKind());
+                                    temps.Add(local.LocalSymbol);
+                                    visitedReceiver = _factory.Sequence(ImmutableArray<LocalSymbol>.Empty, ImmutableArray.Create<BoundExpression>(store), local);
+                                    receiverStored = true;
+                                    break;
+
+                                case >= 0 when positionsAssignedToTemp[argIndex]:
+                                    local = visitedArgumentsBuilder[argIndex] switch
+                                    {
+                                        BoundSequence { Value: BoundLocal l } => l,
+                                        BoundLocal l => l, // Can happen for discard arguments
+                                        var u => throw ExceptionUtilities.UnexpectedValue(u)
+                                    };
+                                    break;
+
+                                case >= 0:
+                                    Debug.Assert(visitedArgumentsBuilder[argIndex] != null);
+                                    var paramIndex = argsToParamsOpt.IsDefault ? argIndex : argsToParamsOpt[argIndex];
+                                    RefKind argRefKind = argumentRefKindsOpt.RefKinds(argIndex);
+                                    RefKind paramRefKind = parameters[paramIndex].RefKind;
+                                    var visitedArgument = visitedArgumentsBuilder[argIndex];
```

Comment:
nit: I think we can assert that previously visited values are not handler placeholders. Consider adding

Reply:
I don't really understand what you're saying here.

Reply:
I'm suggesting `Debug.Assert(visitedArgument  is not BoundInterpolatedStringArgumentPlaceholder)`

Reply:
I guess I don't really know what that assert buys us.

Reply:
In general, placeholders shouldn't survive a visit. BoundInterpolatedStringArgumentPlaceholders are placeholders like any other.